types/src/v0/impls/solver.rs

```diff
@@ -0,0 +1,67 @@
+use committable::{Commitment, Committable};
+use hotshot::types::SignatureKey;
+
+use super::v0_3::{RollupRegistrationBody, RollupUpdatebody};
+
+impl Committable for RollupRegistrationBody {
+    fn tag() -> String {
+        "ROLLUP_REG".to_string()
+    }
+
+    fn commit(&self) -> Commitment<Self> {
+        let mut comm = committable::RawCommitmentBuilder::new(&Self::tag())
+            .u64_field("namespace_id", u64::from(self.namespace_id))
+            .var_size_field("reserve_url", self.reserve_url.as_str().as_ref())
+            .fixed_size_field("reserve_price", &self.reserve_price.to_fixed_bytes())
+            .fixed_size_field("active", &[u8::from(self.active)])
+            .constant_str("signature_keys");
+
+        for key in self.signature_keys.iter() {
+            comm = comm.var_size_bytes(&key.to_bytes());
+        }
+
+        comm = comm
+            .var_size_field("signature_key", &self.signature_key.to_bytes())
+            .var_size_field("text", self.text.as_bytes());
+
+        comm.finalize()
+    }
+}
+
+impl Committable for RollupUpdatebody {
```

Comment:
Here again I think it may be better to have the implementation on the outer type, to avoid exposing the internals.